### PR TITLE
8286707: JFR: Don't commit JFR internal jdk.JavaMonitorWait events

### PR DESCRIPTION
**Pull request by jaokim**
Changed the JFR chunk rotation lock object to specific internal class. This allows that specific Object.wait() event to be skipped, thus not adding JFR internal noise to recordings. 

# Testing
- jdk_jfr

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8286707](https://bugs.openjdk.org/browse/JDK-8286707): JFR: Don't commit JFR internal jdk.JavaMonitorWait events

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Erik Gahlin](https://openjdk.org/census#egahlin) (@egahlin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/8883/head:pull/8883` \
`$ git checkout pull/8883`

Update a local copy of the PR: \
`$ git checkout pull/8883` \
`$ git pull https://git.openjdk.org/jdk pull/8883/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8883`

View PR using the GUI difftool: \
`$ git pr show -t 8883`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/8883.diff">https://git.openjdk.org/jdk/pull/8883.diff</a>

</details>

### CODE REVIEW

File: src/jdk.jfr/share/classes/jdk/jfr/internal/FileDeltaChangeLockObject.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * questions.
+ */
+
+package jdk.jfr.internal;
+
+/**
+ * JFR internal object used for the file delta change locks. A call
+ * to Object.wait() on this object will not generate a JFR event.
+ */
+class FileDeltaChangeLockObject {}
```
Comment (jaokim) [inline src/jdk.jfr/share/classes/jdk/jfr/internal/FileDeltaChangeLockObject.java:32]: Or would a more general lock object be preferred? I couldn't find any more explicitly JFR related monitors in use.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
I think any filtering should be handled internally to JFR not in the ObjectMonitor code.

**Review by egahlin — APPROVED** [review]

**Review by dholmes-ora — COMMENTED** [review]
I am still not happy about this approach. JFR is written in Java and so triggers numerous events that are defined for Java code - that is a simple consequence of implementing it in Java. If there is to be filtering of events that originate from JFR Java code then there should be a general purpose filtering mechanism available to say "exclude JFR code" or "include JFR code" as desired. And to me the filtering mechanism should reside in the event commit code inside the JFR code not in the event-posting code in the shared components of the VM.

**Comment by mgronlun** [comment]
There exist a general exclusion/inclusion mechanism already. But it is an all-or-nothing proposition. This particular case is a thread that we can't exclude because it runs the periodic events, upon being notified. It is the notification mechanism to run the periodic events that trigger this large amount of unnecessary MonitorWait events. Even should we change it to some util.concurrent construct, we are only pushing the problem, because we might be instrumenting them later too. To work with the existing exclusion mechanism, the system would have to introduce an additional thread, which will be excluded, which only handles the notification, and then by some other means triggers another periodic thread (included) to run the periodic events.

**Comment by dholmes-ora** [comment]
@mgronlun my request is that this filtering be done inside the commit logic by the JFR code, not at the site where the event is generated - ie this internal-jfr-event filtering is internalized into the JFR code.

**Comment by jaokim** [comment]
I'm realizing that the monitor class isn't available elsewhere; it is in objectMonitor.cpp, and passed to private members in the `JavaObjectMonitorEvent`, which in turn is generated. The only possibility I see to moving the logic away from objectMonitor.cpp is to change how the event's classes are generated. Either by adding getter to certain members (in this case something like `get_monitorClass `to `JavaObjectMonitorEvent`), or trying to shoehorn the `is_excluded` method into the generated `JavaObjectMonitorEvent`. Neither of these approaches would I say are even worth considering, and they'd also still introduce overhead.

**Review by dholmes-ora — APPROVED** [review]
Thank you for making the effort to investigate alternatives, it is appreciated.

**Pull request closed without merging**